core/gui/src/app/workspace/component/code-editor-dialog/code-editor.component.ts

```diff
@@ -167,28 +168,113 @@ export class CodeEditorComponent implements AfterViewInit, SafeStyle, OnDestroy
     return this.sanitizer.bypassSecurityTrustHtml(textCSS);
   }
 
+  private getFileSuffixByLanguage(language: string): string {
+    switch (language.toLowerCase()) {
+      case "python":
+        return ".py";
+      case "r":
+        return ".r";
+      case "javascript":
+        return ".js";
+      case "java":
+        return ".java";
+      default:
+        return ".py";
+    }
+  }
+
+  private initMonaco(): void {
+    if (this.wrapper) {
+      from(this.wrapper.dispose(true))
+        .pipe(takeUntil(this.componentDestroy))
+        .subscribe({
+          next: () => this.initializeMonacoEditor(),
+        });
+    } else {
+      this.initializeMonacoEditor();
+    }
+  }
+
   /**
    * Create a Monaco editor and connect it to MonacoBinding.
    * @private
    */
-  private initMonaco() {
-    const editor = monaco.editor.create(this.editorElement.nativeElement, {
-      language: this.language,
-      fontSize: 11,
-      theme: "vs-dark",
-      automaticLayout: true,
-    });
-
-    if (this.code) {
-      new MonacoBinding(
-        this.code,
-        editor.getModel()!,
-        new Set([editor]),
-        this.workflowActionService.getTexeraGraph().getSharedModelAwareness()
-      );
+  private initializeMonacoEditor() {
+    if (!this.wrapper && this.code) {
+      const fileSuffix = this.getFileSuffixByLanguage(this.language);
+      this.wrapper = new MonacoEditorLanguageClientWrapper();
+
+      const userConfig: UserConfig = {
+        wrapperConfig: {
+          editorAppConfig: {
+            $type: "extended",
+            codeResources: {
+              main: {
+                text: this.code.toString(),
+                uri: `in-memory-${this.operatorID}.${fileSuffix}`,
```

Comment:
Is it possible to remove this line?

Reply:
I think we might need to keep this line, because the fileSuffix is related to which of the following is being used: import "@codingame/monaco-vscode-python-default-extension"; import "@codingame/monaco-vscode-r-default-extension"; import "@codingame/monaco-vscode-java-default-extension"; And when I tried removing this line, the features like colors stopped working.